app.go

```diff
@@ -198,12 +204,19 @@ func (a *App) Run(arguments []string) (err error) {
 	}
 
 	// Run default Action
-	a.Action(context)
-	return nil
+	err = HandleAction(a.Action, context)
+
+	if err != nil {
+		HandleExitCoder(err)
+	}
+	return err
 }
 
-// Another entry point to the cli app, takes care of passing arguments and error handling
+// DEPRECATED: Another entry point to the cli app, takes care of passing arguments and error handling
 func (a *App) RunAndExitOnError() {
+	fmt.Fprintln(os.Stderr,
```

Comment:
I think deprecating this is great, but I'm a little worried that the end-user (consumer of whatever CLI app uses this library) will see this warning instead of the programmer if they blindly update the library and release. Hopefully this would rarely happen, but maybe we include a note to inform the distributor in case an end-user sees this?